pkg/internal/ebpf/common/pids.go

```diff
@@ -38,26 +38,41 @@ type PIDsFilter struct {
 	pidNs   uint32
 	log     *slog.Logger
 	current map[uint32]struct{}
-	added   chan uint32
-	deleted chan uint32
+	queue   chan pidEvent
+}
+
+type PIDEventOp uint8
+
+const (
+	ADD PIDEventOp = iota + 1
+	DEL
+)
+
+type pidEvent struct {
+	pid uint32
+	op  PIDEventOp
 }
 
 func NewPIDsFilter(log *slog.Logger) *PIDsFilter {
 	return &PIDsFilter{
 		pidNs:   readNamespace(),
 		log:     log,
 		current: map[uint32]struct{}{},
-		added:   make(chan uint32, updatesBufLen),
-		deleted: make(chan uint32, updatesBufLen),
+		queue:   make(chan pidEvent, updatesBufLen),
```

Comment:
I had to merge the two channels, because there were no guarantees that the events will come in order. Essentially the new test failed on my machine, Add, add, delete, the delete event got processed first. I decided to make it one single channel and distinguish the op based on the type, that way it will always happen in order.

Reply:
Yeah I realized this issue too when I saw that unit tests became flaky. Thanks for fixing!